### ISSUE TITLE
FR: Rusage data from previous executed commands?

### ISSUE THREAD

**Issue by gczajkowskiTT** [Rongronggg9/powerline-exectime#4]
Is there are possibility to extend `powerline-exectime` to measure other process performance metrics? For example, I would be very excited about measuring UserTime, CpuTime, and MRSS to start. I am already using some segments for overall machine metrics, but nothing much is available for per command execution metrics?

Implementation wise wrapping all bash commands around /usr/bin/time is one possibility? 

Linux
```
/usr/bin/time  --portability -v sleep 1
	Command being timed: "sleep 1"
	User time (seconds): 0.00
	System time (seconds): 0.00
	Percent of CPU this job got: 0%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 0:01.02
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 2172
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 0
	Minor (reclaiming a frame) page faults: 76
	Voluntary context switches: 78
	Involuntary context switches: 0
	Swaps: 0
	File system inputs: 0
	File system outputs: 0
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 0                              => Available with other segments
```

MacOS instructions retired, cycles elapsed also look intriguing 
```
/usr/bin/time -l -h -p sleep 1                                                                                                                                                               real 1.01
user 0.00
sys 0.00
             1146880  maximum resident set size
                   0  average shared memory size
                   0  average unshared data size
                   0  average unshared stack size
                 186  page reclaims
                   1  page faults
                   0  swaps
                   0  block input operations
                   0  block output operations
                   0  messages sent
                   0  messages received
                   0  signals received
                   0  voluntary context switches
                  20  involuntary context switches
             6579412  instructions retired
             6424272  cycles elapsed
              934592  peak memory footprint
```

**Comment by Rongronggg9** [comment]
Wrapping up every command seems possible using DEBUG trap and `shopt -s extdebug`. However, this is quite complicated and probably messes things up. What's worse, each command in pipelines (`a | b | c`) will be trapped, causing performance drawback. Another major blocker is that some "commands" are in fact Bash builtins and thus cannot be wrapped with `/usr/bin/time` (they can only be wrapped with `time`).

The good news is that we can measure user time and system time by utilizing Bash builtin `times`. I implemented this in 970af6f4.

![user_sys_time](https://github.com/user-attachments/assets/5a9a7a13-26f2-4560-a791-a157f0b4597a)

**Comment by gczajkowskiTT** [comment]
Awesome thank you!
